python/pyspark/sql/readwriter.py

```diff
@@ -563,6 +563,60 @@ def partitionBy(self, *cols):
         self._jwrite = self._jwrite.partitionBy(_to_seq(self._spark._sc, cols))
         return self
 
+    @since(2.3)
+    def bucketBy(self, numBuckets, *cols):
+        """Buckets the output by the given columns on the file system.
```

Comment:
Thank you for adding the wrapper. 

Yes. We should make the Python APIs consistent with Scala APIs, if possible.